src/OpenTelemetry.AutoInstrumentation.Native/startup_hook.h

```diff
@@ -7,22 +7,31 @@
 namespace trace
 {
 
-inline WSTRING GetExpectedStartupHookPath(const WSTRING& home_path) {
-    WSTRING separator = EmptyWStr;
-    if (home_path.back() != DIR_SEPARATOR) {
-        separator = DIR_SEPARATOR;
+inline bool IsStartupHookValid(const std::vector<WSTRING>& startup_hooks, const WSTRING& home_path)
+{
+    if (startup_hooks.empty())
+    {
+        return false;
     }
 
-    return home_path + separator + WStr("netcoreapp3.1") + DIR_SEPARATOR +
-            WStr("OpenTelemetry.AutoInstrumentation.StartupHook.dll");
-}
-
-inline bool IsStartupHookEnabled(const WSTRING& startup_hooks, const WSTRING& home_path)
-{
-    const WSTRING expected_startup_hook = GetExpectedStartupHookPath(home_path);
-    auto n = startup_hooks.find(expected_startup_hook);
+    const auto expected_startuphook_path =
+        std::filesystem::path(home_path)
+        / "netcoreapp3.1" / "OpenTelemetry.AutoInstrumentation.StartupHook.dll";
+    if (!std::filesystem::exists(expected_startuphook_path))
```

Comment:
That is an elegant way to handle to path separators and provide a more robust file system check. Hopefully this approach will also work on Alpine Linux.